File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> storage;
        for (auto num : nums){
            storage[num]++;
        }
        priority_queue<pair<int,int>> order;
        for (auto elem : storage){
            pair<int,int> p = {elem.second, elem.first};
            order.push(p);
        }
        
        vector<int> result;
        for (int i=0; i<k; i++){
            pair<int,int> top = order.top();
            order.pop();
            result.push_back(top.second);
        }
        return result;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        storage = {}
        for num in nums:
            if num in storage:
                storage[num] -= 1
            else:
                storage[num] = -1
        
        heap = []
        for num, freq in storage.items():
            heapq.heappush(heap, (freq,num))
        
        result = []
        for i in range(k):
            freq, num = heapq.heappop(heap)
            result.append(num)
        return result

        